### PR TITLE
updating documentation to common k8s labels

### PR DESCRIPTION
**Pull request by gaerfield**
As suggested by @hzxuzhonghu https://github.com/istio/istio/issues/10724#issuecomment-1272202860 the labels could be updated to k8s common labels

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/deployment/requirements/index.md [hunk 1]
```diff
@@ -71,15 +71,15 @@ To be part of a mesh, Kubernetes pods must satisfy the following requirements:
     Otherwise, you will need to [provide the permission](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#authorizing-policies).
 
 - **Pods with app and version labels**: We recommend adding an explicit
-  `app` label and `version` label to the specification of the pods deployed using
-  a Kubernetes `Deployment`. The `app` and `version` labels add contextual information
+  `app.kubernetes.io/name` label and `app.kubernetes.io/version` label to the specification of the pods deployed using
```
Comment (hzxuzhonghu) [inline content/en/docs/ops/deployment/requirements/index.md:74]: Should we keep `app` and `version` as well, just say they are the second option?

**Comment by windsonsea** [comment]
LGTM for the zh part that is consistent with en. No much text changed.

File: content/en/docs/ops/deployment/requirements/index.md (hunk 1 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/requirements/index.md:74]: Yes, we should mention both labels can be used, but the kubernetes ones are preferred. We should also update all the Istio samples to use the new ones.

Comment (GregHanson) [reply content/en/docs/ops/deployment/requirements/index.md:74]: is there any plan to fully transition to just the Kubernetes app and revision labels?  Will users have to switch to one over the other eventually?

**Review by craigbox — COMMENTED** [review]
Superceded by #14095.

**Pull request closed without merging**